### PR TITLE
VerifyingKey hashing for HashSets

### PR DESCRIPTION
**Pull request by bazzilic**

### CODE REVIEW

File: taiga_halo2/examples/sudoku/app/valid_sudoku/circuit.rs [hunk 1]
```diff
@@ -163,8 +163,18 @@ impl plonk::Circuit<pallas::Base> for SudokuCircuit {
         // The length of u is 41 bytes, or 328 bits, since we are allocating 4 bits
         // per the first 40 integers and let the last sudoku digit to take an entire byte.
         // We still need to add 184 bits (i.e. 23 bytes) to reach 2*256=512 bits in total.
-        let u2 = [u, vec![0; 23]].concat();
-        let u_first: [u8; 32] = u2[0..32].try_into().unwrap();
+        let mut u2 = [0u8; 64];
+        let mut i = 0;
+        let mut j = 0;
+        while j != u.len() {
```
Comment (Acentelles) [inline taiga_halo2/examples/sudoku/app/valid_sudoku/circuit.rs:169]: Could you write a comment about this change?

File: taiga_halo2/examples/sudoku/main.rs [hunk 2]
```diff
@@ -53,6 +53,7 @@ fn main() {
     pub_instance_vec.append(&mut vec_puzzle);
     assert_eq!(
         MockProver::run(13, &circuit, vec![pub_instance_vec.clone()])
+        // TODO:        ^^ should this '13' be 'K'?
```
Comment (Acentelles) [inline taiga_halo2/examples/sudoku/main.rs:56]: Yes

Comment (vveiln) [reply taiga_halo2/examples/sudoku/main.rs:56]: What is K btw? Was just looking at the sudoku code a few days ago and couldn't figure it out

Comment (Acentelles) [reply taiga_halo2/examples/sudoku/main.rs:56]: The size parameter of the circuit (which has 2^k rows).

The more complex the circuit it, the larger this needs to be and the slower it runs

**Comment by bazzilic** [comment]
Hmm... I'm now thinking: did I accidentally implement hashing for the wrong `VerifyingKey` struct? :) @Acentelles

File: taiga_halo2/examples/sudoku/app/valid_sudoku/circuit.rs (hunk 1 above)
Comment (bazzilic) [reply taiga_halo2/examples/sudoku/app/valid_sudoku/circuit.rs:174]: Done.

**Comment by Acentelles** [comment]
Yes, unless we want to change the type in the `application_vp` field in `Note`

File: taiga_halo2/examples/sudoku/keys.rs [hunk 3]
```diff
@@ -18,6 +19,16 @@ impl VerifyingKey {
     }
 }
 
+impl Hash for VerifyingKey {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        let s = format!("{:?}", self.vk.pinned());
+        s.hash(state);
+        let mut v = Vec::new();
+        self.params.write(&mut v); // TODO: properly process result
```
Comment (Acentelles) [inline taiga_halo2/examples/sudoku/keys.rs:27]: I don't think hashing the params is necessary. It suffices with the vk

**Comment by bazzilic** [comment]
Closing this PR in favor of #102 and #107.

**Pull request closed without merging**